src/controllers/scripting/legacy/controllerscriptinterfacelegacy.h

```diff
@@ -17,6 +17,17 @@ class ConfigKey;
 class ControllerScriptInterfaceLegacy : public QObject {
     Q_OBJECT
   public:
+    enum WellKnownCharsets {
```

Comment:
consider using `enum class` and `using enum WellKnownCharsets` to avoid implicit conversions to/from `int`.

Reply:
Wait, where should I use `using enum WellKnownCharsets`?

Reply:
At class scope, usually right below the `enum class` declaration. See [this example](https://compiler-explorer.com/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:102,endLineNumber:10,positionColumn:102,positionLineNumber:10,selectionStartColumn:102,selectionStartLineNumber:10,startColumn:102,startLineNumber:10),source:'struct+A+%7B%0A++++enum+class+Enum+%7B%0A++++++++V1,+V2%0A++++%7D%3B%0A++++using+enum+Enum%3B%0A%7D%3B%0A%0Aint+main()+%7B%0A++++A::Enum+v+%3D+A::V1%3B+//+would+have+to+be+A::Enum::V1+without+the+%60using%60+above.%0A++++return+static_cast%3Cint%3E(v)%3B+//+we+still+retain+the+%22no+implicit+conversions+from/to+int%22+feature.%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'--std%3Dc%2B%2B20',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4). I'm unsure however how well this works with `Q_ENUM`. You'll need to experiment a little whether it can pick up on the `using enum` feature.